src/main/java/com/wasilewskyy/medical_clinic/service/PatientService.java

```diff
@@ -31,7 +35,20 @@ public void deletePatientByEmail(String email) {
     }
 
     public Patient updatePatient(String email, Patient updatedPatient) {
+        Optional<Patient> existingPatient = patientRepository.findByEmail(email);
```

Comment:
Ktos ci mowi ze masz pacjenta o takim mailu zaktualizować -> jesli go nie ma to powinienes po prostu rzucic wyjatek i powiedziec ze takiego pacjenta nie ma bo nie ma sensu dalej wykonywac jakiegokolwiek kodu zwiazane z aktualizacja skoro nie ma kogo aktualizować. 
        Patient existingPatient = patientRepository.findByEmail(email).orElseThrow(() -> new IllegalArgumentException("Patient with given email does not exist"));